### ISSUE TITLE
Empty strings instead of null for missing values for str dtype

### ISSUE THREAD

**Issue by YuRiTan** [pola-rs/polars#878]
#### Are you using Python or Rust?

Replace this text with the **Python**.

#### What version of polars are you using?

`0.8.6`

#### What operating system are you using polars on?

Ubuntu 18.04

#### Describe your bug.

Inconsistent use of missing values. For numeric columns, this is `null`, whereas `str` columns are `""`. Also, the `is_null()` method returns `False` for the empty strings.

#### What are the steps to reproduce the behavior?

There is probably an easier way to reproduce, but this is a MWE close to how I saw this behaviour.

```python
import polars as pl

df = pl.DataFrame({
    'sensor': ['a']*4 + ['b']*5, 
    'should_contain_1_to_10': range(1, 10),
    'other_value': [3, 4, 1, 2 ,3, 5, 1, 2, 3]
})

# remove some rows from the dataframe
df = df.sample(6)

# now I want to fix the column: 'should_contain_0_to_10'
# so I create the full range, and join back to that dataframe

full_range = pl.DataFrame({'should_contain_1_to_10': range(1, 10)})

fixed_df = full_range.join(df, how='left', on='should_contain_1_to_10')

print(fixed_df)
```

```
shape: (9, 3)
╭────────────────────────┬────────┬─────────────╮
│ should_contain_1_to_10 ┆ sensor ┆ other_value │
│ ---                    ┆ ---    ┆ ---         │
│ i64                    ┆ str    ┆ i64         │
╞════════════════════════╪════════╪═════════════╡
│ 1                      ┆ "a"    ┆ 3           │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 2                      ┆ "a"    ┆ 4           │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 3                      ┆ ""     ┆ null        │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 4                      ┆ "a"    ┆ 2           │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ ...                    ┆ ...    ┆ ...         │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 5                      ┆ "b"    ┆ 3           │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 6                      ┆ "b"    ┆ 5           │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 7                      ┆ ""     ┆ null        │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 8                      ┆ ""     ┆ null        │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 9                      ┆ "b"    ┆ 3           │
```

**Comment by YuRiTan** [comment]
The behaviour seems to change with kernel restarts. Sometimes is shows `null` and sometimes the above described behaviour.

**Comment by YuRiTan** [comment]
When rerunning multiple times, it sometimes even shows both:

```
shape: (9, 3)
╭────────────────────────┬────────┬─────────────╮
│ should_contain_1_to_10 ┆ sensor ┆ other_value │
│ ---                    ┆ ---    ┆ ---         │
│ i64                    ┆ str    ┆ i64         │
╞════════════════════════╪════════╪═════════════╡
│ 1                      ┆ "a"    ┆ 3           │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 2                      ┆ ""     ┆ null        │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 3                      ┆ "a"    ┆ 1           │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 4                      ┆ "a"    ┆ 2           │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ ...                    ┆ ...    ┆ ...         │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 5                      ┆ "b"    ┆ 3           │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 6                      ┆ **""**     ┆ null        │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 7                      ┆ "b"    ┆ 1           │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 8                      ┆ "b"    ┆ 2           │
├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 9                      ┆ **null**   ┆ null        │
╰────────────────────────┴────────┴─────────────╯

```

**Comment by ritchie46** [comment]
Fixed in #876
